### DISCUSSION TITLE
I think this is more correct solution for gig pots 12th July 2023 solution

### DISCUSSION THREAD

**Discussion by jatinavi** [getlost01/gfg-potd#6] [Solution Ideas] [answered]
class Solution {
public:
    const int mod = 1000000007; // Define the modulus value
    
    long long powerexpo(long long a, long long n) {
        long long res = 1; // Use long long to avoid overflow
        while (n) {
            if (n & 1) {
                res = (res % mod * a % mod) % mod;
                --n;
            } else {
                a = (a % mod * a % mod) % mod;
                n >>= 1;
            }
        }
        return res;
    }

    long long power(int N, int R) {
        return powerexpo(N, R);
    }
};

**Comment by getlost01** [comment, marked as answer]
Ya, thanks for pointing these issues.

1. Regarding the comment `long long res = 1; // Use long long to avoid overflow`, it is indeed a mistake that I acknowledge.
2. However, during code submission, `# define mod 1000000007` this modulus value is already defined by default in the driver code, that why I didn't redefine it. But now I think they might changed the driver code.

Just so you know, I only upload my solution after successfully passing all the test cases with the code.
